model/vm_host.rb

```diff
@@ -243,6 +244,56 @@ def reset
     Hosting::Apis.reset_server(self)
   end
 
+  def check_storage_smartctl(ssh_session, devices)
+    devices.map { |device_name| ssh_session.exec!("sudo smartctl -j -H /dev/#{device_name} | jq .smart_status.passed").strip == "true" }.all?(true)
+
+  end
+
+  def check_storage_nvme(ssh_session, devices)
+    devices.map { |device_name| ssh_session.exec!("sudo nvme smart-log /dev/#{device_name} | grep \"critical_warning\" | awk '{print $3}'").strip == "0" }.all?(true)
+  end
+
+  def check_storage_read_write(ssh_session, devices)
+    parser = LsblkParser.new(ssh_session.exec!("lsblk --json"))
+    device_mount_points = devices.map { |device| parser.get_device_mount_points(device)}
+
+    all_mount_points = []
+    device_mount_points.each do |device_mount_point|
+      device_mount_point.each do |device_name, mount_points|
+        mount_points.each do |mount_point|
+          all_mount_points.append(mount_point)
+        end
+      end
+    end
+
+    success = true
+    all_mount_points.uniq.each do |mount_point|
+      file_name = mount_point == "/" ? "/test-file" : "#{mount_point}/test-file"
+      ssh_session.exec!("sudo bash -c \"head -c 1M </dev/zero > #{file_name}\"")
+      success = false unless ssh_session.exec!("sha256sum #{file_name}").strip == "30e14955ebf1352266dc2ff8067e68104607e750abb9d3b36582b8af909fcb58  #{file_name}"
+      ssh_session.exec!("sudo rm #{file_name}")
+    end
+    success
+
+  end
+
+  def check_storage_kernel_logs(ssh_session, devices)
+    kernel_logs = ssh_session.exec!("journalctl -kS -5min --no-pager").split("\n")
```

Comment:
`-5min` is somehow important here. That means, in case of a repair, we would detect unavailable for 5 more minutes. Maybe we can reduce this to 1min for now?

Reply:
good point. I reduced it to 1min.